bom/build.gradle.kts

```
@@ -21,3 +45,109 @@ dependencies {
+            assert(dependencyManagementNode != null) {
+                "<dependencyManagement> node not found in the generated BOM POM"
+            }
+            val dependenciesNode = dependencyManagementNode.getNode("dependencies")
+            assert(dependenciesNode != null) { "<dependencies> node not found in the generated BOM POM" }
+
+            val existingScalaDeps =
+                dependenciesNode!!
+                    .children()
+                    .map { it as Node }
+                    .filter { it.getNode("artifactId")?.text()?.contains("scala") ?: false }
+
+            existingScalaDeps.forEach {
+                val groupId: String = it.getNode("groupId")!!.text()
+                val originalArtifactId: String = it.getNode("artifactId")!!.text()
+                val artifactVersion: String = it.getNode("version")!!.text()
+
+                // Add multiple versions with Scala suffixes for each Scala-related dependency.
+                scalaVersions!!.forEach { scalaVersion ->
+                    if (scalaVersion != defaultScalaVersion) {
+                        // Replace scala version suffix
+                        val newArtifactId: String = originalArtifactId.replace(defaultScalaVersion, scalaVersion)
+                        val dependencyNode = dependenciesNode.appendNode("dependency")
+                        dependencyNode.appendNode("groupId", groupId)
+                        dependencyNode.appendNode("artifactId", newArtifactId)
+                        dependencyNode.appendNode("version", artifactVersion)
+                    }
+                }
+            }
+        }
+    }
+}
+
+/*
+ * Validate the BOM file.
+ */
+tasks.withType<GenerateMavenPom> {
+    doLast {
+        pom.withXml {
+            val pomXml: Node = asNode()
+            val dependenciesNode = pomXml.getNode("dependencyManagement").getNode("dependencies")
+            assert(dependenciesNode!!.children().isNotEmpty()) {
```

These assertions only take effect when the JVM is run with the -ea flag, which could lead to silent failures. We could replace `assert()` with `require()` to ensure the validation always runs.